controllers/cluster_controller.go

```diff
@@ -65,41 +64,37 @@ const (
 // ClusterReconciler reconciles a Cluster object
 type ClusterReconciler struct {
 	Client client.Client
-	Log    logr.Logger
 
-	scheme          *runtime.Scheme
 	restConfig      *rest.Config
 	recorder        record.EventRecorder
 	externalTracker external.ObjectTracker
 }
 
-func (r *ClusterReconciler) SetupWithManager(mgr ctrl.Manager, options controller.Options) error {
+func (r *ClusterReconciler) SetupWithManager(ctx context.Context, mgr ctrl.Manager, options controller.Options) error {
 	controller, err := ctrl.NewControllerManagedBy(mgr).
 		For(&clusterv1.Cluster{}).
 		Watches(
 			&source.Kind{Type: &clusterv1.Machine{}},
-			&handler.EnqueueRequestsFromMapFunc{ToRequests: handler.ToRequestsFunc(r.controlPlaneMachineToCluster)},
+			handler.EnqueueRequestsFromMapFunc(r.controlPlaneMachineToCluster),
 		).
 		WithOptions(options).
-		WithEventFilter(predicates.ResourceNotPaused(r.Log)).
+		WithEventFilter(predicates.ResourceNotPaused(ctrl.LoggerFrom(ctx))).
 		Build(r)
 
 	if err != nil {
 		return errors.Wrap(err, "failed setting up with a controller manager")
 	}
 
 	r.recorder = mgr.GetEventRecorderFor("cluster-controller")
-	r.scheme = mgr.GetScheme()
 	r.restConfig = mgr.GetConfig()
 	r.externalTracker = external.ObjectTracker{
 		Controller: controller,
 	}
 	return nil
 }
 
-func (r *ClusterReconciler) Reconcile(req ctrl.Request) (_ ctrl.Result, reterr error) {
-	ctx := context.Background()
-	logger := r.Log.WithValues("cluster", req.Name, "namespace", req.Namespace)
+func (r *ClusterReconciler) Reconcile(ctx context.Context, req ctrl.Request) (_ ctrl.Result, reterr error) {
+	log := ctrl.LoggerFrom(ctx)
```

Comment:
As per the documentation for `ctrl.LoggerFrom(ctx)`, it seems like it auto fills in the `req.Name` and `req.Namespace` which is great! 

However, it feels like some of the readability is lost because before we had a log with a key `cluster`. Also, it seems we are also removing the `WithName` options when setting up the logger in the reconcilers.
If these logs were aggregated, it may be difficult to read. 
_I'm assuming that the context keys used are generically `name` and `namespace`. I didn't get a chance to dig that deep._

Maybe we can do something like this once in every respective reconciler.
```suggestion
	log := ctrl.LoggerFrom(ctx).WithName("controllers").WithName("Cluster")
```
WDYT?

Reply:
I believe https://github.com/kubernetes-sigs/controller-runtime/blob/c000ea850121b53dd769f624f3bfa74a4fcf100f/pkg/controller/controller.go#L129 sets up the controller name that you're looking for, for all controllers.

And yes, the keys are namespace and name: https://github.com/kubernetes-sigs/controller-runtime/blob/c000ea850121b53dd769f624f3bfa74a4fcf100f/pkg/internal/controller/controller.go#L98.

I can see where you're coming from comparing
```
"namespace"="foo" "cluster"="bar"
```
to
```
"namespace"="foo" "name"="bar"
```
but if we also consider the controller name, does it make it any better?
```
"controller"="cluster" "namespace"="foo" "name"="bar"
```
or perhaps this?
```
"controller"="cluster" "reconcile-request"="foo/bar"
```

Reply:
Thanks for going through the codebase @ncdc :), there is also https://github.com/kubernetes-sigs/controller-runtime/blob/c000ea850121b53dd769f624f3bfa74a4fcf100f/pkg/builder/controller.go#L253-L257 which sets up the `reconcilerGroup` and `reconcilerKind` in the controller logger

Reply:
We should be able to bring some changes upstream if the logs don't fit nicely with our expectations, there is still time until v0.7.0 is released

Reply:
@ncdc @vincepri Thanks for pointing out the name of the controller and reconciler's GVK being added to the logger. I'm not sure where or how our logger is being threaded in, because our current usage is `ctrl.LoggerFrom` which calls https://github.com/kubernetes-sigs/controller-runtime/blob/c000ea850121b53dd769f624f3bfa74a4fcf100f/pkg/log/log.go#L57 which seems to be returning a new logger. 
~_Unless there is some shared logger that is being used within controller-runtime._~
UPDATE: I see that this logger is added into the context. https://github.com/kubernetes-sigs/controller-runtime/blob/c000ea850121b53dd769f624f3bfa74a4fcf100f/pkg/internal/controller/controller.go#L99


I haven't had the chance yet to deploy the branch and view the logs directly myself so it might look fine. This was just something I thought I'd bring to attention but don't believe it's a blocker.

Reply:
We should definitely revisit / review the logs after this merges, let's open a different issue? I do want to have consistent logging everywhere as part of the v0.4.0 release, and everything should use the same logger and have consistent names

Reply:
I'm fine with opening a different issue to track this work and ensure that logs are legible and useful to the reader. That seems like a sizable chunk of work anyways.

Reply:
cluster controller logs now look like:

```
controller/cluster "msg"="Reconciliation is paused for this object" "name"="bmo" "namespace"="default" "reconcilerGroup"="cluster.x-k8s.io" "reconcilerKind"="Cluster"
```